packages/eslint-plugin/src/rules/prefer-literal-enum-member.ts

```diff
@@ -59,16 +59,46 @@ export default createRule({
           return;
         }
 
+        // Allow bitwise operators when both operands are literals or refer to another member on this enum
         if (
           allowBitwiseExpressions &&
           node.initializer.type === AST_NODE_TYPES.BinaryExpression &&
           ['|', '&', '^', '<<', '>>', '>>>'].includes(
             node.initializer.operator,
           ) &&
-          node.initializer.left.type === AST_NODE_TYPES.Literal &&
-          node.initializer.right.type === AST_NODE_TYPES.Literal
+          node.parent.type === AST_NODE_TYPES.TSEnumDeclaration
         ) {
-          return;
+          const leftIsLiteral =
+            node.initializer.left.type === AST_NODE_TYPES.Literal;
+          const rightIsLiteral =
+            node.initializer.right.type === AST_NODE_TYPES.Literal;
+
+          // early branch, so gathering member identifiers only happens when needed
+          if (leftIsLiteral && rightIsLiteral) {
+            return;
+          }
+
+          const members = node.parent.members.reduce((names, member) => {
+            if ('name' in member.id) {
```

Comment:
[Code] We generally try not to use `'in'` checks or similar for AST code. If a property (here, `id`) doesn't exist it's because the node legitimately might be a type that doesn't have it. See test cases across the enum-related rules for property names based on literals:

```ts
enum Example {
  ['abc'] = 0,
}
```

The code here should be able to handle names like that, including test coverage.